nomad/structs/services.go

```diff
@@ -515,6 +515,9 @@ type ConsulConnect struct {
 
 	// SidecarService is non-nil if a service requires a sidecar.
 	SidecarService *ConsulSidecarService
+
+	// SidecarTask is non-nil if sidecar overrides are set
+	SidecarTask *Task
```

Comment:
Validate? (or is that later?)

Reply:
Yeah we have #6038 and I'll probably do some validation as part of the admission controller.